### ISSUE TITLE
Incorrect output shape for logits output returned from get_output("logits")

### ISSUE THREAD

**Issue by VishalX** [microsoft/onnxruntime-genai#1385]
**Describe the bug**
I am trying to check perplexity of Phi3.5 model I exported using Olive (See Example linked below). This is a quantized with QDQ nodes and have static shapes (sliding window of 64) with decoder-pipeline.

When I call get_output("logits"), It always returns tensor with [64, N] regardless of the input prompt length.

The logits shape is correct if I use a regular gen-ai model without sliding window..

**To Reproduce**
1. Use olive example to export the model (https://github.com/microsoft/Olive/tree/main/examples/phi3_5#qdq-model-with-4-bit-weights--16-bit-activations)
2. Run the model (something like below)
      ```py
      seqlen = 128
      input_tokens = test_encodings[ : seqlen] 
      print("Input Seq length: ", input_tokens.shape)
      
      generator = og.Generator(model, params)
      generator.append_tokens(input_tokens)
      
      logits =  torch.tensor(generator.get_output("logits"))
      print("- Get Output:", logits.shape)
      ```
Outputs:
```sh
- Get Output: (1, 64, 32064)
```

**Expected behavior**

For Phi3.5, the logits shape should be (1, 128, 32064) for input sequence length of 128. 

**Desktop (please complete the following information):**
 - OS: [Windows]

**Comment by VishalX** [comment]
@ramkrishna2910

**Comment by baijumeswani** [comment]
The model you're trying to run uses a window size of 64 tokens. So, for a prompt length of 128 tokens, the context/prompt model is executed twice in chunks of 64 sequence lengths. When you call `generator.append_tokens(input_tokens)`, ort-genai will execute the model twice (64 + 64). Each model execution results in a `logits` of shape`(1, 64, 32064)` as you've found. 

If you would like to get the complete logits, you could send in window size tokens at a time (with a new generator for each chunk), get the logits output, and concatenate the logits for each append tokens call. This is not ideal, but we don't have a good way to get all the logits at the moment since we do not keep all the logits in memory for past session run calls.

**Comment by VishalX** [comment]
Thanks for the quick response @baijumeswani. I can try the the concat way for now.
Any plans to enable this inside OGA in future, maybe with an option or flag in genai_config if the concern in memory utilization or perf?

**Comment by baijumeswani** [comment]
We don't have any plans as of now to support this scenario. If you'd like to contribute, we'll be happy to review and accept the changes.

**Comment by ramkrishna2910** [comment]
@baijumeswani, exposing logprobs for the entire prompt in windowed execution is critical. Most accuracy tests require logprobs for evaluation. 
We can try the concat method in our application, but getting logprobs from OGA would be ideal.
Can we please keep this issue open ? :)

**Comment by Henry0215** [comment]
> The model you're trying to run uses a window size of 64 tokens. So, for a prompt length of 128 tokens, the context/prompt model is executed twice in chunks of 64 sequence lengths. When you call `generator.append_tokens(input_tokens)`, ort-genai will execute the model twice (64 + 64). Each model execution results in a `logits` of shape`(1, 64, 32064)` as you've found.
> 
> If you would like to get the complete logits, you could send in window size tokens at a time (with a new generator for each chunk), get the logits output, and concatenate the logits for each append tokens call. This is not ideal, but we don't have a good way to get all the logits at the moment since we do not keep all the logits in memory for past session run calls.

Hi  @baijumeswani , I have different opinion on this issue. I don't think it will work wll due to the causality of LLMs. When doing model prefilling, the model will get "Causal Mask" to let the subsequent tokens obtain previous tokens' information (the past tokens' K and V projections). But your solution seems to ignore the casuality for the second generator will not get the first generator's KV cache. I think the generator class needs to provide a new function to allow forwarding a sequence with the past key values. Meanwhile, I have an idea about Speculative Decoding, which needs this function as well. I would be glad if you could fix this issue, and can provide some support. Thanks.
